### PR TITLE
mlflow 2.18.0: make a multi-output recipe

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-6051](https://anaconda.atlassian.net/browse/PKG-6051) 
- [Upstream repository](https://github.com/mlflow/mlflow/tree/v2.18.0)
- Requirements:
  - https://github.com/mlflow/mlflow/blob/v2.18.0/pyproject.skinny.toml
  - https://github.com/mlflow/mlflow/blob/v2.18.0/requirements/skinny-requirements.txt
  - https://github.com/mlflow/mlflow/blob/v2.18.0/pyproject.toml 
  - https://github.com/mlflow/mlflow/blob/v2.18.0/requirements/core-requirements.txt

### Explanation of changes:

- Rewriting as a multi-output recipe because of catching a [conda-build bug](https://github.com/conda/conda-build/issues/5416) on **win-64** (see the previous draft PR https://github.com/AnacondaRecipes/mlflow-feedstock/pull/8), the fix won't be available until the next release of conda-build in January 2025 (???), see https://github.com/conda/conda-build/pull/5417
  - conda-build 24.11.2 is expected to fix the issue https://github.com/conda/conda-build/issues/5524 
- Make changes based on the conda-forge recipe:
  - Add README_SKINNY.rst 
  - Add build-mlflow.bat and build-mlflow.sh
  - Remove conda_build_config.yaml as we do not rely on jinja2 variants and selectors.
- Skip py<39
- Add to `run_constrained`:
  -  `mlflow <0a0` for the `mlflow-skinny` subpackage to prevent installing `mlflow` to the same environment
  -  `mlflow-skinny <0a0` for the `mlflow` subpackage to prevent installing `mlflow-skinny` to the same environment
- Add separate `about` sections to every output. They have different summaries and descriptions

### Notes:

-

[PKG-6051]: https://anaconda.atlassian.net/browse/PKG-6051?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,197 +1,267 @@
...
+        - mlflow=mlflow.cli:cli
+    requirements:
+      host:
+        - pip
+        - python
+        - setuptools
+        - wheel
+      run:
+        - python
+        - cachetools<6,>=5.0.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - cachetools <6,>=5.0.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,197 +1,267 @@
...
+        - click >=7.0,<9
+        - cloudpickle <4
+        - databricks-sdk <1,>=0.20.0
+        - gitpython >=3.1.9,<4
+        - importlib-metadata <9,>=3.7.0,!=4.7.0
+        - opentelemetry-api<3,>=1.9.0
+        - opentelemetry-sdk<3,>=1.9.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
        - opentelemetry-api <3,>=1.9.0
        - opentelemetry-sdk <3,>=1.9.0
```

**Review by cbouss — APPROVED** [review]
LGTM, with some pedantic suggestions.

**Comment by skupr-anaconda** [comment]
@cbouss I'll wait for conda-build 24.11.2 to be deployed on Prefect. Today, there is still 24.11.1. Then, this PR will be superseded by https://github.com/AnacondaRecipes/mlflow-feedstock/pull/8 where changes are minimal.

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

**Pull request closed without merging**